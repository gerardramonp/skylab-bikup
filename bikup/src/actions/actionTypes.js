export default {
    LOAD_BIKES: 'LOAD_BIKES',
    LOAD_BIKE_BY_ID: 'LOAD_BIKE_BY_ID'
};
