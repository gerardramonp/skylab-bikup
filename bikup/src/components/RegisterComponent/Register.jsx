import React from 'react';
import { NavLink, useHistory } from 'react-router-dom';

import './Register.scss';

function Register() {
    const history = useHistory();

    function handleLogoClick() {
        history.push('/');
    }

    return (
        <>
            <div className="landing__header desktop">
                <img
                    className="logo"
                    src="https://cdn.discordapp.com/attachments/692420285143711814/693437226146594876/LogoGerili.png"
                    alt="logo"
                />
                <p className="appName">bikUP</p>
                <div className="flex-spacer"></div>
                <button className="register__button register__button--header">
                    Log In
                </button>
            </div>
            <div className="register__upper mobile">
                <img
                    src="https://cdn.discordapp.com/attachments/692420285143711814/693437226146594876/LogoGerili.png"
                    alt="bikup logo"
                    className="register__logo"
                    onClick={handleLogoClick}
                />
                <NavLink to="/" className="register__bikup">
                    bikUP
                </NavLink>
                <div className="flex-spacer"></div>
                <NavLink className="login__button" to="/login">
                    Log In
                </NavLink>
            </div>
            <div className="register">
                <div className="register__container">
                    <h1>Create an account</h1>

                    <form action="" className="register__form" method="POST">
                        <label htmlFor="email" className="input__label">
                            Email
                        </label>
                        <input
                            type="email"
                            className="register__email register__input"
                            name="email"
                            placeholder="Enter your email..."
                        />
                        <label htmlFor="password" className="input__label">
                            Password
                        </label>
                        <input
                            type="password"
                            className="register__password register__input"
                            name="password"
                            placeholder="Enter your password..."
                        />
                        <label
                            htmlFor="repeat-password"
                            className="input__label"
                        >
                            Repeat Password
                        </label>
                        <input
                            type="password"
                            className="register__repeat-password register__input"
                            name="repeat-password"
                            placeholder="Repeat the password..."
                        />
                        <div className="login__warning warning__error"></div>
                        <div className="register__form__bottom">
                            <button className="register__button" type="submit">
                                Create Account
                            </button>
                        </div>
                    </form>

                    <div className="register__separator">
                        <div className="line"></div>
                        <span className="bold">Or</span>
                        <div className="line"></div>
                    </div>

                    <div className="register__external">
                        <button className="register__btn strava">
                            <div className="btn__logo">
                                <img
                                    src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53a18bf970328231db4f61/2f6af477333d585dfbfc19a14da9f857/strava-2.svg"
                                    alt="google logo"
                                />
                            </div>
                            <div className="btn__text">
                                register with Strava
                            </div>
                        </button>

                        <button className="register__btn google">
                            <div className="btn__logo">
                                <img
                                    src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53a18bf970328231db4f61/99c6b9ab576aa9e324980db26e5fc7c0/google-logo.png"
                                    alt="google logo"
                                />
                            </div>
                            <div className="btn__text">
                                register with Google
                            </div>
                        </button>
                    </div>
                </div>
            </div>
        </>
    );
}

export default Register;
