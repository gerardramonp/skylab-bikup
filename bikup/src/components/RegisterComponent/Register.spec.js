import React from 'react';
import renderer from 'react-test-renderer';
import Register from './Register';
import { BrowserRouter as Router } from 'react-router-dom';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import dispatcher from '../../dispatcher';
import actionTypes from '../../actions/actionTypes';

Enzyme.configure({ adapter: new Adapter() });

describe('Profile Component', () => {
	let tree = null;
	let compo = null;
	let mountedCompo = null;

	beforeEach(() => {
		tree = renderer.create(
			<Router>
				<Register />
			</Router>
		);
	});

	it('should match', () => {
		dispatcher.dispatch({
			type: actionTypes.LOGIN_USER_STRAVA,
			data: true,
		});
		expect(tree.toJSON()).toMatchSnapshot();
	});

	it('should find button', () => {
		const page = shallow(<Register />);
		const button = page.find('.form-register__button');
		button.simulate('click', { preventDefault: () => {} });
		expect(button.length).toBe(1);
	});
});
