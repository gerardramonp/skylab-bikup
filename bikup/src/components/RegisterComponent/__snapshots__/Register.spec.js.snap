// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Component should match 1`] = `
Array [
  <div
    className="landing__header desktop"
  >
    <img
      alt="logo"
      className="logo"
      src="https://cdn.discordapp.com/attachments/692420285143711814/693437226146594876/LogoGerili.png"
    />
    <p
      className="appName"
    >
      bikUP
    </p>
    <div
      className="flex-spacer"
    />
    <button
      className="register__button register__button--header"
    >
      Log In
    </button>
  </div>,
  <div
    className="register__upper mobile"
  >
    <img
      alt="bikup logo"
      className="register__logo"
      onClick={[Function]}
      src="https://cdn.discordapp.com/attachments/692420285143711814/693437226146594876/LogoGerili.png"
    />
    <a
      aria-current="page"
      className="register__bikup active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      bikUP
    </a>
    <div
      className="flex-spacer"
    />
    <a
      aria-current={null}
      className="login__button"
      href="/login"
      onClick={[Function]}
    >
      Log In
    </a>
  </div>,
  <div
    className="register"
  >
    <div
      className="register__container"
    >
      <h1>
        Create an account
      </h1>
      <form
        action="/api/auth/mail"
        className="register__form"
        method="POST"
        name="registerForm"
      >
        <label
          className="input__label"
          htmlFor="email"
          required={true}
        >
          Email
        </label>
        <input
          className="register__email register__input"
          name="email"
          placeholder="Enter your email..."
          required={true}
          type="email"
        />
        <label
          className="input__label"
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="register__password register__input"
          name="password"
          pattern="^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%^&*_=+-]).{8,12}$"
          placeholder="Enter your password..."
          required={true}
          type="password"
        />
        <label
          className="input__label"
          htmlFor="repeat-password"
        >
          Repeat Password
        </label>
        <input
          className="register__repeat-password register__input"
          name="repeat_password"
          onChange={[Function]}
          pattern="^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%^&*_=+-]).{8,12}$"
          placeholder="Repeat the password..."
          required={true}
          type="password"
        />
        <div
          className="login__warning "
        />
        <div
          className="register__form__bottom"
        >
          <button
            className="form-register__button"
            onClick={[Function]}
            type="submit"
          >
            <div
              className="loading hidden"
            >
              <img
                alt="loading..."
                className="loading__img"
                src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f5753c458a8b552f891bb81/af512a8cb3c1285000d1191fdaaa670c/Spinner-1s-200px_(1).gif"
              />
              <p>
                Loading...
              </p>
            </div>
            <div
              Account={true}
              Create={true}
              className="no-loading"
            >
              Create an account
            </div>
          </button>
        </div>
      </form>
      <div
        className="register__separator"
      >
        <div
          className="line"
        />
        <span
          className="bold"
        >
          Or
        </span>
        <div
          className="line"
        />
      </div>
      <div
        className="register__external"
      >
        <button
          className="register__btn strava"
        >
          <div
            className="btn__logo"
          >
            <img
              alt="google logo"
              src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53a18bf970328231db4f61/2f6af477333d585dfbfc19a14da9f857/strava-2.svg"
            />
          </div>
          <div
            className="btn__text"
          >
            register with Strava
          </div>
        </button>
        <button
          className="register__btn google"
        >
          <div
            className="btn__logo"
          >
            <img
              alt="google logo"
              src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53a18bf970328231db4f61/99c6b9ab576aa9e324980db26e5fc7c0/google-logo.png"
            />
          </div>
          <div
            className="btn__text"
          >
            register with Google
          </div>
        </button>
      </div>
    </div>
  </div>,
]
`;
